bentoml/_internal/io_descriptors/json.py

```diff
@@ -248,8 +248,15 @@ async def from_http_request(self, request: Request) -> JSONType:
     async def to_http_response(
         self, obj: JSONType | pydantic.BaseModel, ctx: Context | None = None
     ):
-        if isinstance(obj, pydantic.BaseModel):
-            obj = obj.dict()
+        try:
+            import pydantic
+
+            # This is to prevent cases where custom JSON encoder is used.
+            if isinstance(obj, pydantic.BaseModel):
```

Comment:
use `if self._pydantic_model`

Reply:
This is not correct right? checking object instance is independent of whether `self._pydantic_model` is set or not.